packages/amazon-sumerian-hosts-core/src/core/awspack/LexUtils.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-underscore-dangle */
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+/**
+ * A collection of useful lex related functions.
+ *
+ * @hideconstructor
+ */
+class LexUtils {
+  /**
+   * Downsamples the audio to a target sample rate.
+   *
+   * Inspired by the following blog post from the Lex team:
+   * https://aws.amazon.com/blogs/machine-learning/capturing-voice-input-in-a-browser/
+   *
+   * @param {Float32Array} buffer - Input audio buffer
+   * @param {float} sourceSampleRate - Sample rate of the input audio buffer
+   * @param {float} targetSampleRate - Sample rate to try to convert to
+   *
+   * @return {Float32Array} Downsampled audio buffer
+   */
+  static downsampleAudio(buffer, sourceSampleRate, targetSampleRate) {
+    if (!buffer || !buffer.length) {
+      return;
+    }
+
+    if (sourceSampleRate === targetSampleRate) {
+      return buffer;
+    }
+
+    if (sourceSampleRate < targetSampleRate) {
+      throw Error(`Input Sample rate ${sourceSampleRate} is less than target sample rate ${targetSampleRate}.`);
+    }
+
+    const bufferLength = buffer.length;
+    const sampleRateRatio = sourceSampleRate / targetSampleRate;
+    const newLength = Math.round(bufferLength / sampleRateRatio);
+
+    const downsampledBuffer = new Float32Array(newLength);
+
+    let position = 0;
+    let bufferOffset = 0;
+    while (position < newLength) {
+      const nextBufferOffset = Math.round((position + 1) * sampleRateRatio);
+
+      let accumulator = 0;
+      let count = 0;
+      for (let i = bufferOffset; i < nextBufferOffset && i < bufferLength; i++) {
+        accumulator += buffer[i];
+        count++;
+      }
+
+      downsampledBuffer[position] = accumulator / count;
+      position++;
+      bufferOffset = nextBufferOffset;
+    }
+
+    return downsampledBuffer;
+  }
+
+  /**
+   * Converts audio data to WAV.
+   *
+   * Inspired by the following blog post from the Lex team:
+   * https://aws.amazon.com/blogs/machine-learning/capturing-voice-input-in-a-browser/
+   *
+   * @param {Float32Array} buffer - Input audio buffer
+   * @param {float} targetSampleRate - Sample rate for the output audio
+   *
+   * @return {DataView} Converted audio data
+   */
+  static encodeWAV(buffer, targetSampleRate) {
```

Comment:
is there a test for this code? This seems risky, as we could break this logic without knowing. Could we have a test with a small hardcoded input and expected output that we could compare against as a minimum?